retrofit/src/main/java/retrofit2/OkHttpCall.java

```diff
@@ -119,22 +117,19 @@ private okhttp3.Call getRawCall() throws IOException {
   public void enqueue(final Callback<T> callback) {
     Objects.requireNonNull(callback, "callback == null");
 
-    okhttp3.Call call;
-    Throwable failure;
-
-    synchronized (this) {
-      if (executed) throw new IllegalStateException("Already executed.");
-      executed = true;
-
-      call = rawCall;
-      failure = creationFailure;
-      if (call == null && failure == null) {
-        try {
-          call = rawCall = createRawCall();
-        } catch (Throwable t) {
-          throwIfFatal(t);
-          failure = creationFailure = t;
-        }
+    if (!executed.compareAndSet(false, true)) throw new IllegalStateException("Already executed.");
+
+    okhttp3.Call call = rawCallRef.get();
+    Throwable failure = creationFailureRef.get();
+
+    if (call == null && failure == null) {
+      try {
+        call = createRawCall();
+        rawCallRef.compareAndSet(null, call);
```

Comment:
This code is now racy. Another thread could be performing the same operation in parallel and its `okhttp3.Call` might "win" causing this instance to be a duplicate.